### PR TITLE
Create nvqc_launch.sh script

### PR DESCRIPTION
**Pull request by bmhowe23**
We should run `cudaq-qpud` as "nobody". With that change, it was getting a bit unwieldy to generate `launch.sh` and then launch it, so I pulled the logic into a standalone script.

### CODE REVIEW

File: scripts/nvqc_launch.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+cat /opt/nvidia/cudaq/build_info.txt
```
Comment (bmhowe23) [inline scripts/nvqc_launch.sh:9]: Most of this script is moving and unquoting commands from docker/release/cudaq.nvqc.Dockerfile into this script. I will highlight the one important change below.

File: scripts/nvqc_launch.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# (2) NVCF function must set container environment variable `NVQC_REST_PAYLOAD_VERSION` equal
+# to the RestRequest payload version with which `cudaq-qpud` in the deployment Docker image was compiled.
+# Failure to do so will result in early exits of the entry point command, thus deployment failure.
+EXPECTED_REST_PAYLOAD_VERSION="$(cudaq-qpud --type nvcf --schema-version | grep -o "CUDA-Q REST API version: \S*" | cut -d ":" -f 2 | tr -d " ")"
+if [[ "$NVQC_REST_PAYLOAD_VERSION" !=  "$EXPECTED_REST_PAYLOAD_VERSION" ]]; then
+  echo "Invalid Deployment: NVQC_REST_PAYLOAD_VERSION environment variable ($NVQC_REST_PAYLOAD_VERSION) does not match cudaq-qpud (expected $EXPECTED_REST_PAYLOAD_VERSION)."
+  exit 1
+fi
+
+python3 /nvqc_scripts/nvqc_proxy.py &
```
Comment (bmhowe23) [inline scripts/nvqc_launch.sh:27]: Path change here

File: scripts/nvqc_launch.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+NUM_ACTUAL_GPUS=$(nvidia-smi --list-gpus | wc -l)
+if [[ "$NUM_GPUS" == "$NUM_ACTUAL_GPUS" ]]; then
+  cd /tmp
+  CMDSTR="mpiexec -np $NUM_ACTUAL_GPUS cudaq-qpud --type nvcf --port 3031"
+  while true; do
+    echo "export PATH=${PATH}; $CMDSTR" | sudo su -s /bin/bash nobody
```
Comment (bmhowe23) [inline scripts/nvqc_launch.sh:34]: Run as nobody (instead of cudaq)

**Comment by bmhowe23** [comment]
Ping for review...

**Review by 1tnguyen — APPROVED** [review]
👍

**Pull request merged**